[Workers] Add clarification on route case sensitivity (#11246)
cloudflare/cloudflare-docs: content/workers/configuration/routing/routes.md

- Route pattern matching considers the entire request URL, including the query parameter string. Since route patterns may not contain query parameters, the only way to have a route pattern match URLs with query parameters is to terminate it with a wildcard, `*`.

- The path component of route patterns is case sensitive, for example, `example.com/Images/*` and `example.com/images/*` are two distinct routes.

> - For routes created prior to October 15th, 2023, the host component of route patterns is case sensitive, for example, `example.com/*` and `Example.com/*` are two distinct routes.

Reviewer: ```suggestion
- For routes created before October 15th, 2023, the host component of route patterns is case sensitive, for example, `example.com/*` and `Example.com/*` are two distinct routes.
```
replace complex vocabulary for better readability

- Route pattern matching considers the entire request URL, including the query parameter string. Since route patterns may not contain query parameters, the only way to have a route pattern match URLs with query parameters is to terminate it with a wildcard, `*`.

- The path component of route patterns is case sensitive, for example, `example.com/Images/*` and `example.com/images/*` are two distinct routes.

> - The host component of route patterns is not case sensitive, for example, `example.com/*` and `Example.com/*` are equivalent routes.

Reviewer 2: Do we want to mention this only applies for new deploys post 15th Oct
Not that it should really matter but since this is a change in behaviour, always good to note.

Author: Technically this only applies to routes created after the 15th. Redeploying a worker with older routes won't change the behavior of those older routes, since we date/version routes individually.

Author: I added another commit that includes the API date cutoff, and agree that this is good to note.
